src/generators/rust/renderers/EnumRenderer.ts

```diff
@@ -43,26 +43,47 @@ ${additionalContent}`;
   runStructMacroPreset(): Promise<string> {
     return this.runPreset('structMacro');
   }
+
+  /**
+   * Returns the type for the JSON value
+   */
+  renderEnumValueType(value: any): string {
+    switch (typeof value) {
+    case 'boolean':
+      return 'bool';
+    case 'bigint':
+      return 'i64';
+    case 'number': {
+      return 'f64';
+    }
+    case 'object': {
+      return 'HashMap<String, serde_json::Value>';
+    }
+    case 'string':
+    default: {
+      return 'String';
+    }
+    }
+  }
 }
 
 export const RUST_DEFAULT_ENUM_PRESET: EnumPresetType<RustOptions> = {
   self({ renderer }) {
     return renderer.defaultSelf();
   },
-  item({ item }) {
-    if (item.value === 'HashMap<String, serde_json::Value>') {
-      return `${item.key}(${item.value})`;
+  item({ item, renderer }) {
+    const typeOfEnumValue = renderer.renderEnumValueType(item.value);
+    if (typeOfEnumValue === 'HashMap<String, serde_json::Value>') {
+      return `${item.key}(${typeOfEnumValue})`;
```

Comment:
I put together a playground to help clarify when ``${item.key}(${typeOfEnumValue})` needs to be returned vs. `${item.key}`

Have a noodle around and let me know what you think: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5c82a5c0219f7e77f74f6f93e2e58208

Reply:
I gotta try using the Rust models for a project at some point! 😄